starfish/pipeline/spots/detector/_base.py

```diff
@@ -1,21 +1,23 @@
-from typing import Callable, Sequence, Union
+from typing import Callable, Sequence, Tuple, Union
 
 import numpy as np
 import xarray as xr
 
 from starfish._stack import ImageStack
 from starfish.intensity_table import IntensityTable
 from starfish.pipeline.algorithmbase import AlgorithmBase
-from starfish.pipeline.features.spot_attributes import SpotAttributes
+from starfish.pipeline.spots.spot_attributes import SpotAttributes
+from starfish.pipeline.spots.detector.combine_adjacent_features import (
+    ConnectedComponentDecodingResult
+)
 from starfish.types import Number
 
 
 class SpotFinderAlgorithmBase(AlgorithmBase):
     def find(
             self,
             hybridization_image: ImageStack,
-            blobs_image: Union[np.ndarray, xr.DataArray]
-    ) -> IntensityTable:
+    ) -> Union[IntensityTable, Tuple[IntensityTable, ConnectedComponentDecodingResult]]:
```

Comment:
WHOA what?  this seems really bad.  is this an intermediate step before we normalize?

Reply:
I hope so. Deep and I need to talk about how the PixelSpotFinder should work.